internal/controller/stackrunjob_controller.go

```diff
@@ -72,15 +74,24 @@ func (r *StackRunJobReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 	// Update step statuses, i.e., when stack run was successful or failed.
 	for _, step := range stackRun.Steps {
```

Comment:
Oh we should actually not do these updates here (let the harness fully own them and it isn't going to impact anything on the api if they remain pending)

Reply:
done